Formula 1/ViewControllers/CircuitsCollectionViewController.swift

```diff
@@ -0,0 +1,69 @@
+//
+//  TracksViewController.swift
+//  Formula 1
+//
+//  Created by Gio on 11/12/19.
+//  Copyright © 2019 Gio. All rights reserved.
+//
+
+import Foundation
+import UIKit
+import Combine
+import SwiftUI
+
+class CircuitsCollectionViewController: UIViewController {
+    @ObservedObject var viewModel = CircuitsViewModel()
+    
+    var collectionView = CircuitsCollectionView()
+    
+    override func viewDidLoad() {
+        title = "Circuits"
+        
+        addSubviews()
+        activateConstraints()
+        
+        enableDelegates()
+    }
+    
+    func addSubviews() {
+        view.addSubview(collectionView)
+    }
+    
+    func activateConstraints() {
+        collectionView.translatesAutoresizingMaskIntoConstraints = false
+        
+        NSLayoutConstraint.activate([
+            collectionView.topAnchor.constraint(equalTo: view.topAnchor),
+            collectionView.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+            collectionView.trailingAnchor.constraint(equalTo: view.trailingAnchor),
+            collectionView.bottomAnchor.constraint(equalTo: view.bottomAnchor)
+        ])
+    }
+    
+    func enableDelegates() {
+        collectionView.delegate = self
+        collectionView.dataSource = self
+        
+        viewModel.delegate = self
+    }
+}
+
+extension CircuitsCollectionViewController: UICollectionViewDelegate, UICollectionViewDataSource {
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int { viewModel.numberOfCircuits }
```

Comment:
Nope. You can commit return but don't make lines long. The shorter the more readable.
```suggestion
    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int { 
        return viewModel.numberOfCircuits 
    }
```

Reply:
perdoname i got carried away - i just recently switched to a smaller scale setting for the monitor